utils/genIconUrl.js

```diff
@@ -64,29 +64,45 @@ async function getIconUrl(skylink) {
       return;
     }
 
-    // Grab the relative path to the manifest file from the manifest link
-    // tag
-    const manifestPath = manifestTag.getAttribute("href");
+    // Use URL to get the manifestPath as this will clean it. We don't grab
+    // the url created as it doesn't include the skylink.
+    const { pathname: manifestPath } = new URL(
+      manifestTag.getAttribute("href"),
+      skylinkUrl
+    );
+
+    // Create the manifestUrl
+    const manifestUrl = skylinkUrl + manifestPath;
 
-    // Generate the manifestUrl and query the manifest
-    const manifestUrl = skylinkUrl + "/" + manifestPath.replace("./", "");
-    const manifestResponse = await got(manifestUrl);
     // Grab the json of the manifest
+    const manifestResponse = await got(manifestUrl);
     const manifest = JSON.parse(manifestResponse.body);
+
     // Grab the iconPath or the src path of the first icon listed if they
     // are present
     const icon = manifest.icons[0].src || manifest.iconPath || undefined;
     if (!icon) {
       return;
     }
-    // Generate the iconPath
-    const iconPath = icon.replace("./", "");
+
+    // Generate the iconUrl and iconPath
+    const { href: iconUrl, pathname: iconPath } = new URL(icon, manifestUrl);
+
+    // Check if the skylink is in the iconPath, for some reason it is not
+    // always present. I'm not sure if it is because of some characters in
+    // the skylink impacting the new URL method.
```

Comment:
if you don't want to include skynet-js then we probably shouldn't use `href` prop and just create iconUrl ourselves with `${portal}${skylink}${iconPath}`

Reply:
Follow Up?

If I'm understanding correctly this can be cleaned up afterwards, since it isn't that we are generating any bad iconUrls. Any bad URLs would be caught by line 102.   

The decision here is to either leave as is or use skynet-js right? I can update the comment with what you wrote in a follow up.  Since it is such a small script it is nice to have as few dependencies as possible imo so if it works to not use skynet-js then I think these extra 5 lines are acceptable.